src/renderer/src/routes/pages/Wallets/CommonAccountActions.tsx

```diff
@@ -24,24 +25,46 @@ export const CommonAccountActions = ({ account }: TProps) => {
   const { t } = useTranslation('common', { keyPrefix: 'general' })
   const { t: tWallets } = useTranslation('pages', { keyPrefix: 'wallets' })
   const balanceQuery = useBalance(account)
+  const unclaimedQuery = useUnclaimed(account)
   const { canMigrateToNeo3 } = useMigrationNeo3Validations({ account })
   const { modalNavigate } = useModalNavigate()
 
+  const service = bsAggregator.blockchainServicesByName[account.blockchain]
+
   const tokenBalances = balanceQuery.data?.tokensBalances ?? []
+  const unclaimedResult = unclaimedQuery.data
+
   const isShowedMigrationNeo3 = account.blockchain === 'neoLegacy'
 
   const isDisabledMigrationNeo3 =
-    isShowedMigrationNeo3 && (balanceQuery.isLoading || !canMigrateToNeo3({ tokenBalances }))
+    isShowedMigrationNeo3 &&
+    (balanceQuery.isLoading || unclaimedQuery.isLoading || !canMigrateToNeo3({ tokenBalances }))
 
   const isSwapAvailable = !!SWAP_NETWORK_BY_BLOCKCHAIN_AND_NETWORK_ID[account.blockchain][network.id]?.length
 
   const handleMigrate = async () => {
-    const service = bsAggregator.blockchainServicesByName[account.blockchain]
+    await UtilsHelper.sleep(100)
+
+    const normalizedFeeTokenHash = UtilsHelper.normalizeHash(service.feeToken.hash)
+    const feeTokenBalance = tokenBalances.find(
+      ({ token }) => UtilsHelper.normalizeHash(token.hash) === normalizedFeeTokenHash
+    )
+
+    if (
+      !!unclaimedResult &&
+      unclaimedResult.unclaimedNumber > 0 &&
+      unclaimedResult.unclaimedNumber > unclaimedResult.feeNumber &&
+      (unclaimedResult.feeNumber === 0 ||
+        (!!feeTokenBalance && feeTokenBalance.amountNumber > unclaimedResult.feeNumber))
+    ) {
```

Comment:
You are doing the same conditional that you did inside `useMigrationNeo3Validation`, what do you think about creating a function there and using it here?